addons/l10n_gr_edi/models/account_move.py

```diff
@@ -0,0 +1,718 @@
+            vat_vals['vat_category'] = 8
+
+        if vat_vals['vat_category'] == 7:
+            # need vat exemption category
+            vat_vals['vat_exemption_category'] = line.l10n_gr_edi_tax_exemption_category
+
+        return vat_vals
+
+    @api.model
+    def _l10n_gr_edi_get_classification_vals(self, line, net_amount):
+        """
+        Returns additional income/expense classification items ("icls"/"ecls") if needed for the detail values.
+        The returned format is: {'ecls': [ {'category': <str>, 'type': <str>, 'amount': <float>}, ... ], 'icls': <same_as_ecls> }
+        :param account.move.line line:
+        :param float net_amount:
+        :rtype: dict[str, list[dict]]
+        """
+        cls_vals = {'ecls': [], 'icls': []}
+
+        if line.l10n_gr_edi_cls_category:
+            cls_vals_list = cls_vals['ecls'] if line.l10n_gr_edi_cls_category in CLASSIFICATION_CATEGORY_EXPENSE else cls_vals['icls']
+            cls_type = line.l10n_gr_edi_cls_type or ''
+            if len(cls_type) > 0 and cls_type[0] == 'X':  # handle duplicate E3 type on inv type 17.5
+                cls_type = cls_type[1:]
+
+            cls_vals_list.append({
+                'category': line.l10n_gr_edi_cls_category,
+                'type': cls_type,
+                'amount': net_amount,
+            })
+            if line.l10n_gr_edi_cls_vat:
+                cls_vals_list.append({
+                    'category': '',
+                    'type': line.l10n_gr_edi_cls_vat,
+                    'amount': net_amount,
+                })
+
+        return cls_vals
+
+    @api.model
+    def _l10n_gr_edi_get_sum_classification_vals(self, details):
+        """
+        Aggregates the amounts from all common classifications from the ``details`` list vals.
+        :param list[dict] details:
+        :rtype: dict[str, list[dict]]
+        """
+        icls_vals, ecls_vals = {}, {}
+        summary_icls, summary_ecls = [], []
+
+        for detail in details:
+            icls_list, ecls_list = detail['icls'], detail['ecls']
+            for icls in icls_list:
+                category_type = (icls['category'], icls['type'])
+                icls_vals.setdefault(category_type, 0)
+                icls_vals[category_type] += icls['amount']
+            for ecls in ecls_list:
+                category_type = (ecls['category'], ecls['type'])
+                ecls_vals.setdefault(category_type, 0)
+                ecls_vals[category_type] += ecls['amount']
+
+        for category_type, total_amount in icls_vals.items():
+            category, cls_type = category_type
+            summary_icls.append({'type': cls_type, 'category': category, 'amount': total_amount})
+        for category_type, total_amount in ecls_vals.items():
+            category, cls_type = category_type
+            summary_ecls.append({'type': cls_type, 'category': category, 'amount': total_amount})
```

Comment:
Here is the rounded amount without tax per line. So except if it has to be different from `summary_total_net_value`, it won't work in round globally. Does it has to be strictly the sum of the amounts from the lines? Usually, the `amount` from the lines are not rounded but the total amount for the whole document is.
To handle that, we usually use a new grouping function. It depends of what we expect here exactly